content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,711 @@
+
+The `llmobs` SDK provides a corresponding inline method to automatically trace the operation a given code block entails. These methods have the same argument signature as their function wrapper counterparts, with the addition that `name` is required, as the name cannot be inferred from an anonymous callback. This method will finish the span under the following conditions:
+
+1. The function returns a Promise, in which case the span will finish when the promise is resolved or rejected.
+2. The function takes a callback as its last parameter, in which case the span will finish when that callback is called.
+3. The function doesn't accept a callback and doesn't return a Promise, in which case the span will finish at the end of the function execution.
+
+#### Example without a callback
+
+{{< code-block lang="javascript" >}}
+function processMessage () {
+  return llmobs.trace({ kind: 'workflow', name: 'processMessage', sessionId: '<SESSION_ID>', mlApp: '<ML_APP>' }, workflowSpan => {
+    ... // user application logic
+    return
+  })
+}
+{{< /code-block >}}
+
+#### Example with a callback
+
+{{< code-block lang="javascript" >}}
+function processMessage () {
+  return llmobs.trace({ kind: 'workflow', name: 'processMessage', sessionId: '<SESSION_ID>', mlApp: '<ML_APP>' }, (workflowSpan, cb) => {
+    ... // user application logic
+    let maybeError = ...
+    cb(maybeError) // the span will finish here, and tag the error if it is not null or undefined
+    return
+  })
+}
+{{< /code-block >}}
+
+The return type of this function will match the return type of the traced function:
+
+{{< code-block lang="javascript" >}}
+function processMessage () {
+  const result = llmobs.trace({ kind: 'workflow', name: 'processMessage', sessionId: '<SESSION_ID>', mlApp: '<ML_APP>' }, workflowSpan => {
+    ... // user application logic
+    return 'hello world'
+  })
+
+  console.log(result) // 'hello world'
+  return result
+}
+{{< /code-block >}}
+
+### Function decorators in TypeScript
+
+The Node.js LLM Observability SDK offers an `llmobs.decorate` function which serves as a function decorator for TypeScript applications. This functions tracing behavior is the same as `llmobs.wrap`.
+
+#### Example
+  
+{{< code-block lang="javascript" >}}
+// index.ts
+import tracer from 'dd-trace';
+tracer.init({
+  llmobs: {
+    mlApp: "<YOUR_ML_APP_NAME>",
+  },
+});
+
+const { llmobs } = tracer;
+
+class MyAgent {
+  @llmobs.decorate({ kind: 'agent' })
+  async runChain () {
+    ... // user application logic
+    return
+  }
+}
+
+{{< /code-block >}}
+
+### Force flushing in serverless environments
+
+`llmobs.flush()` is a blocking function that submits all buffered LLM Observability data to the Datadog backend. This can be useful in serverless environments to prevent an application from exiting until all LLM Observability traces are submitted.
+
+### Tracing multiple applications
+
+The SDK supports tracking multiple LLM applications from the same service.
```

Comment:
Heading says "tracing", first sentence says "tracking" — which one is correct?

Reply:
should probably be "tracing" - will update!